staging/src/k8s.io/apimachinery/pkg/runtime/error.go

```diff
@@ -71,11 +72,11 @@ func (k *notRegisteredErr) Error() string {
 // IsNotRegisteredError returns true if the error indicates the provided
 // object or input data is not registered.
 func IsNotRegisteredError(err error) bool {
+	var nerr *notRegisteredErr
 	if err == nil {
 		return false
 	}
-	_, ok := err.(*notRegisteredErr)
```

Comment:
nit: `errors.As` catches cases where type assertion doesn't. A little concerned that people may get unexpected behavior if they call this method first, then do a type assertion themselves.

Reply:
tests for these would be good to make sure we're using errors.As correctly... something like 

```go
func TestIsNotRegisteredError(t *testing.T) {
	testcases := []struct {
		name   string
		err    error
		expect bool
	}{
		{name: "nil", err: nil, expect: false},
		{name: "other", err: fmt.Errorf("different error"), expect: false},
		{name: "direct", err: NewNotRegisteredErrForKind("test", schema.GroupVersionKind{}), expect: true},
		{name: "embedded", err: fmt.Errorf("embedded: %w", NewNotRegisteredErrForKind("test", schema.GroupVersionKind{})), expect: true},
	}
	for _, tc := range testcases {
		t.Run(tc.name, func(t *testing.T) {
			if actual := IsNotRegisteredError(tc.err); actual != tc.expect {
				t.Fatalf("expected %v, got %v", tc.expect, actual)
			}
		})
	}
}
```

Reply:
> tests for these would be good to make sure we're using errors.As correctly... something like
> 
> ```go
> func TestIsNotRegisteredError(t *testing.T) {
> 	testcases := []struct {
> 		name   string
> 		err    error
> 		expect bool
> 	}{
> 		{name: "nil", err: nil, expect: false},
> 		{name: "other", err: fmt.Errorf("different error"), expect: false},
> 		{name: "direct", err: NewNotRegisteredErrForKind("test", schema.GroupVersionKind{}), expect: true},
> 		{name: "embedded", err: fmt.Errorf("embedded: %w", NewNotRegisteredErrForKind("test", schema.GroupVersionKind{})), expect: true},
> 	}
> 	for _, tc := range testcases {
> 		t.Run(tc.name, func(t *testing.T) {
> 			if actual := IsNotRegisteredError(tc.err); actual != tc.expect {
> 				t.Fatalf("expected %v, got %v", tc.expect, actual)
> 			}
> 		})
> 	}
> }
> ```

thx, updated